functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+			Container: pod.Spec.Containers[0].Name,
+			Command:   command,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+	retry := 3
+	for retry > 0 {
+		exec, err := remotecommand.NewSPDYExecutor(client.Client.Config, "POST", req.URL())
+		Expect(err).ToNot(HaveOccurred())
+		var buf bytes.Buffer
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: &buf,
+			Stderr: os.Stderr,
+			Tty:    true,
+		})
+		if err == nil {
+			commandOut := buf.String()
+			return commandOut, buf
+		}
+		log.Print(err)
+		retry--
+		time.Sleep(time.Second)
+	}
+	return "", bytes.Buffer{}
+}
+
+func compareOffsetTime(timeDiff string) bool {
+	var timeStampList []int
+	for _, line := range strings.Split(timeDiff, "\n") {
+		if strings.Contains(line, "openshift_ptp_max_offset_from_master") && !strings.Contains(line, "# ") {
+			splitedLine := strings.Split(line, " ")
+			offcetFromMaster, err := strconv.Atoi(strings.Trim(splitedLine[len(splitedLine)-1], "\r"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(offcetFromMaster).To(BeNumerically("<", 100))
+			timeStampList = append(timeStampList, offcetFromMaster)
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+func getPodLogs(pod v1core.Pod) (string, error) {
+	podLogOpts := v1core.PodLogOptions{}
+	req := client.Client.CoreV1Interface.Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+
+	return str, nil
+}
+
+func netParseEthtoolTimeStampFeature(cmdOut *bytes.Buffer) bool {
+	var hardRxEnabled bool
+	var hardTxEnabled bool
+	var hardRawEnabled bool
+
+	glog.V(2).Infof("cmd output for %v", cmdOut)
+	scanner := bufio.NewScanner(cmdOut)
+	for scanner.Scan() {
+		line := strings.TrimPrefix(scanner.Text(), "\t")
+		parts := strings.Fields(line)
+		if parts[0] == _ETHTOOL_HARDWARE_RECEIVE_CAP {
+			hardRxEnabled = parts[1] == _ETHTOOL_RX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_TRANSMIT_CAP {
+			hardTxEnabled = parts[1] == _ETHTOOL_TX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_RAW_CLOCK_CAP {
+			hardRawEnabled = parts[1] == _ETHTOOL_RAW_HARDWARE_FLAG
+		}
+	}
+	return hardRxEnabled && hardTxEnabled && hardRawEnabled
+}
+
+func ptpDisoveredInterfaceList(path string) []string {
+	var ptpInterfaces []string
+	fg, err := client.Client.CoreV1Interface.RESTClient().Get().AbsPath(path).DoRaw()
+	Expect(err).ToNot(HaveOccurred())
+	var nodePtpDevice ptpv1.NodePtpDevice
+	err = json.Unmarshal(fg, &nodePtpDevice)
+	Expect(err).ToNot(HaveOccurred())
+	for _, intConf := range nodePtpDevice.Status.Devices {
+		ptpInterfaces = append(ptpInterfaces, intConf.Name)
+	}
+	return ptpInterfaces
+}
+
+func detectDefaultRouteInterface(routeTable string) (string, error) {
```

Comment:
Please add a comment on top of this function to explain how the routeTable looks on what you do here.

It is hard from the if to understand.